infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -290,23 +302,113 @@ public void reboot() {
                 .<LogAddress,LogData>weigher((k, v) -> v.getData() == null ? 1 : v.getData().readableBytes())
                 .maximumWeight(maxCacheSize)
                 .removalListener(this::handleEviction)
-                .writer(new CacheWriter<LogAddress, LogData>() {
```

Comment:
I'm still not happy with removing the use of Caffeine here, since I'd rather rely on Caffeine's atomicity guarantees. I think you need to switch LoadingCache to AsyncLoadingCache, which will give you a Java8 CompletableFuture on a get. You can then use that to return a response asynchronously, by chaining the completion of the cache retrieval, and handling the exception by sending the appropriate error as well.